neo/SmartContract/Native/Tokens/NeoToken.cs

```diff
@@ -0,0 +1,422 @@
+                    break;
+                case "initialize":
+                    result = Initialize(engine);
+                    break;
+                case "unclaimedGas":
+                    result = UnclaimedGas(engine, args[0].GetByteArray(), (uint)args[1].GetBigInteger());
+                    break;
+                case "registerValidator":
+                    result = RegisterValidator(engine, args[0].GetByteArray());
+                    break;
+                case "vote":
+                    result = Vote(engine, args[0].GetByteArray(), ((VMArray)args[1]).Select(p => p.GetByteArray().AsSerializable<ECPoint>()).ToArray());
+                    break;
+                case "getValidators":
+                    result = GetValidators(engine).Select(p => (StackItem)p.ToArray()).ToArray();
+                    break;
+                default:
+                    return false;
+            }
+            engine.CurrentContext.EvaluationStack.Push(result);
+            return true;
+        }
+
+        private static StorageKey CreateStorageKey(UInt160 script_hash, byte prefix, byte[] key = null)
+        {
+            StorageKey storageKey = new StorageKey
+            {
+                ScriptHash = script_hash,
+                Key = new byte[sizeof(byte) + (key?.Length ?? 0)]
+            };
+            storageKey.Key[0] = prefix;
+            if (key != null)
+                Buffer.BlockCopy(key, 0, storageKey.Key, 1, key.Length);
+            return storageKey;
+        }
+
+        private static BigInteger BalanceOf(ApplicationEngine engine, byte[] account)
+        {
+            if (account.Length != 20) throw new ArgumentException();
+            StorageItem storage = engine.Service.Snapshot.Storages.TryGet(CreateStorageKey(ScriptHash, Prefix_Account, account));
+            if (storage is null) return BigInteger.Zero;
+            Struct state = (Struct)storage.Value.DeserializeStackItem(engine.MaxArraySize);
+            return state[0].GetBigInteger();
+        }
+
+        private static bool Transfer(ApplicationEngine engine, byte[] from, byte[] to, BigInteger amount)
+        {
+            if (engine.Service.Trigger != TriggerType.Application) throw new InvalidOperationException();
+            UInt160 hash_from = new UInt160(from);
+            UInt160 hash_to = new UInt160(to);
+            if (amount.Sign < 0) throw new ArgumentOutOfRangeException(nameof(amount));
+            if (!hash_from.Equals(new UInt160(engine.CurrentContext.CallingScriptHash)) && !engine.Service.CheckWitness(engine, new UInt160(from)))
+                return false;
+            ContractState contract_to = engine.Service.Snapshot.Contracts.TryGet(hash_to);
+            if (contract_to?.Payable == false) return false;
+            StorageKey key_from = CreateStorageKey(ScriptHash, Prefix_Account, from);
+            StorageItem storage_from = engine.Service.Snapshot.Storages.TryGet(key_from);
+            if (amount.IsZero)
```

Comment:
Why to proceed normally and return `true` when amount is zero (and then reaching 192)?